documentation/coding_guidelines.md

```diff
@@ -167,9 +167,7 @@ We recommend that you use `//` and `///` comments rather than block comments `/*
 
 ### Cloning
 
-If `x` is reference counted, prefer [`Arc::clone(x)`](https://doc.rust-lang.org/std/sync/struct.Arc.html) over `x.clone()`. [`Arc::clone(x)`](https://doc.rust-lang.org/std/sync/struct.Arc.html) explicitly indicates that we are cloning `x`. This avoids confusion about whether we are performing an expensive clone of a `struct`, `enum`, other types, or just a cheap reference copy.
-
-Also, if you are passing around [`Arc<T>`](https://doc.rust-lang.org/std/sync/struct.Arc.html) types, consider using a newtype wrapper:
+If you are passing around [`Arc<T>`](https://doc.rust-lang.org/std/sync/struct.Arc.html) types, consider using a newtype wrapper:
```

Comment:
why btw?

Reply:
This is legacy guidelines, I would argue it is good practice to wrap arc + locks into a new type if it improves utility of the code.

Reply:
Any reason to change preferring Arc::clone(x) vs x.clone()?

Reply:
too many chars to type (that's my reason)

Reply:
So the extra 5 chars is not worth the clarity? :). Anything else?

Reply:
I don't think the use of Arc::clone is systemic through the code base and there are no lints for it. If we have a lint, then I'm all game, but as it is, it is often ignored. Also I don't really know why this should really matter? We should prefer references if we can use references, except maybe on copyable types. If we have to clone, we have to clone.

Reply:
I think in general, there doesn't appear to be a huge gain one way or another but rather we should aim for consistency in the codebase as a principle - even if there isn't a lint rule.  I'm okay either way as long as we pick one way. =)

Reply:
I'm happy to run a sed script to remove them. But I think what folks want to do is really up to them. Are we going to say eliminate the use of Type::clone() from the code base? It is syntactically correct rust. Similarly should we insist that you cannot use a trait directly with the type as an input but rather must run the trait via an instance of a struct that implements that trait (generalizing the Arc argument to all structs)?

Reply:
I don't see a strong argument for those broader suggestions.  I thought the existing text was a reasonable justification for preferring Arc::clone(). It's also used quite a bit in our current codebase (100+ times).

Reply:
I'm willing to bet there are more instances of instance::clone() where instance is of type Arc<T> then there are of Arc::clone(instance)... I've added quite a lot myself. A guideline that isn't easy to follow or enforce and has little ramifications seems superfluous and distracts from guidelines we really care about. I'd be perfectly fine with striking this whole section since there are examples of folks passing around Arc<T> instead of T { innner: Arc<S> }. I digress.

Reply:
I agree with a guideline that is hard to follow is not very useful.  As I mentioned above, enforcement is a nice to have but not required for a guideline to be useful.

Since it is a bit tricky to figure out when to Arc::clone() vs clone() in downstream cases as well as @zekun000's argument to reduce chars, I'm fine with this change.